### PR TITLE
Fix BWUtil report on early exit

### PR DESCRIPTION
**Pull request by gonidelis**
## Description

We used to read all the elements to report `BWUtil` even on early exit for `thrust::equal`. That results on `BWUtil` being reported to be more than 100% in cases of early exit. Random example:

```
 |  BWUtil  |
 |----------|
 |    2.67% |
 |   27.06% |
 |   31.98% |
 |   32.49% |
 |    2.71% |
 |   27.20% |
 |   56.16% |
 |   64.26% |
 |    2.70% |
 |   27.25% |
 |  441.01% |  <----
 | 7039.20% |  <----
```

This fixes that.

### CODE REVIEW

**Comment by bernhardmgruber** [comment]
I noticed this as well when I added the benchmark in #1944 and added a question for the reviewers: https://github.com/NVIDIA/cccl/pull/1944#pullrequestreview-2160391767. However, both @miscco and @gevtushenko did not react to the comment and approved the PR, so I eventually dropped my worries.

Maybe they missed the commend, so it's good you bring this up again!

**Comment by bernhardmgruber** [comment]
I would actually favor not reporting a bandwidth at all. It just highly depends on the implementation of `thrust::equals` how many data is read. For example, it is not specified whether we start comparing from the beginning, end, in the middle, or even every n-th block, etc. All of those implementations would be valid.

**Comment by gonidelis** [comment]
@bernhardmgruber  We do have a `CommonPrefixRatio` counter that expresses how much of prefix is actually equal percentage wise. Isn't that what you are talking about?

**Comment by gevtushenko** [comment]
> However, both @miscco and @gevtushenko did not react to the comment and approved the PR

My bad, I forgot to answer your question.  

> I would actually favor not reporting a bandwidth at all. 

I'm not sure if tuning infrastructure and CI scripts support that. 

> For example, it is not specified whether we start comparing from the beginning, end, in the middle, or even every n-th block, etc.

For now, all the execution policies in Thrust read from the beginning and short circuit. We can revisit this question when we get one of the implementations that do not read from the beginning.

File: thrust/benchmarks/bench/equal/basic.cu [hunk 1]
```diff
@@ -21,7 +21,10 @@ static void benchmark(nvbench::state& state, nvbench::type_list<T>)
   thrust::fill(policy(alloc), b.begin() + different_elems, b.end(), T{2});
 
   state.add_element_count(elements);
-  state.add_global_memory_reads<T>(2 * elements);
+  state.add_global_memory_reads<T>(std::max(2 * different_elems, std::size_t(1))); // using `different_elements` instead
```
Comment (gevtushenko) [inline thrust/benchmarks/bench/equal/basic.cu:24]: suggestion: I suppose we have to read two elements to compare them:
```suggestion
  state.add_global_memory_reads<T>(2 * std::max(different_elems, std::size_t(1))); // using `different_elements` instead
```

File: thrust/benchmarks/bench/equal/basic.cu [hunk 2]
```diff
@@ -21,7 +21,10 @@ static void benchmark(nvbench::state& state, nvbench::type_list<T>)
...
+                                                                                   // of `elements` corresponds to the
+                                                                                   // actual elements read in an early
+                                                                                   // exit
   state.exec(nvbench::exec_tag::no_batch | nvbench::exec_tag::sync, [&](nvbench::launch&) {
     do_not_optimize(thrust::equal(policy(alloc), a.begin(), a.end(), b.begin()));
```
Comment (gevtushenko) [inline thrust/benchmarks/bench/equal/basic.cu:29]: suggestion: while we are here, can you also fix this?
```suggestion
  state.exec(nvbench::exec_tag::no_batch | nvbench::exec_tag::sync, [&](nvbench::launch& launch) {
    do_not_optimize(thrust::equal(policy(alloc, launch), a.begin(), a.end(), b.begin()));
```

**Review by gevtushenko — APPROVED** [review]

**Review by bernhardmgruber — APPROVED** [review]
The name `different_elems` confused me, because it seems to me it actually denotes the number of equal elements. If you agree, please also rename the variable, e.g. to `same_elems`.

It could be that I wanted to name the variable `different_elems_start`, but somehow failed to complete the name :)

Also, the variable `name` at the top is no longer needed. Please also remove it. Thx!

**Comment by bernhardmgruber** [comment]
> > I would actually favor not reporting a bandwidth at all.
> 
> I'm not sure if tuning infrastructure and CI scripts support that.

Good point! Also, if we report no bandwidth, we cannot report a SOL%, right?

**Review by bernhardmgruber — APPROVED** [review]

**Pull request merged**